File: Coding Challenges/Daily Challenge/2026/January/20260117-Knight-Moves.py
def knight_moves(position):
    possible_moves = [[-2,1],[-2,-1],[-1,2],[-1,-2],[1,2],[1,-2],[2,1],[2,-1]]
    columns = "ABCDEFGH"
    old_coords = [
        columns.index(position[:1]),
        int(position[1:]) - 1
    ]
    possible_pos = []

    for move in possible_moves:
        new_pos = [old_coords[0] + move[0], old_coords[1] + move[1]]

        if new_pos[0] >= 0 and new_pos[0] < 8 and new_pos[1] >= 0 and new_pos[1] < 8:
            possible_pos.append(columns[new_pos[0]] + str(new_pos[1] + 1))


    return len(possible_pos)

File: Coding Challenges/Daily Challenge/2026/January/20260117-Knight-Moves.js
function knightMoves(position) {
  const possibleMoves = [[-2,1],[-2,-1],[-1,2],[-1,-2],[1,2],[1,-2],[2,1],[2,-1]];
  const columns = "ABCDEFGH";
  const oldCoords = [
    columns.indexOf(position.slice(0,1)),
    Number(position.slice(1)) - 1
  ];
  let possiblePos = []

  for(let i = 0; i < possibleMoves.length; i++) {
    let move = possibleMoves[i];
    let newPos = [oldCoords[0] + move[0], oldCoords[1] + move[1]];

    if(newPos[0] >= 0 && newPos[0] < 8 && newPos[1] >= 0 && newPos[1] < 8) {
      possiblePos.push(columns[newPos[0]] + (newPos[1] + 1));      
    }
  }

  return possiblePos.length;
}